chain-extensions/xvm/src/lib.rs

```diff
@@ -76,28 +72,22 @@
                 // So we will charge a 32KB dummy value as a temporary replacement.
                 let charged_weight = env.charge_weight(weight_limit.set_proof_size(32 * 1024))?;
 
-                let caller = match env.ext().caller().clone() {
-                    Origin::Signed(address) => address,
-                    Origin::Root => {
-                        log::trace!(
-                            target: "xvm-extension::xvm_call",
-                            "root origin not supported"
-                        );
-                        return Ok(RetVal::Converging(
-                            XvmExecutionResult::from(CallError::BadOrigin).into(),
-                        ));
-                    }
-                };
+                let XvmCallArgs {
+                    vm_id,
+                    to,
+                    input,
+                    value,
+                } = env.read_as_unbounded(env.in_len())?;
 
-                let XvmCallArgs { vm_id, to, input } = env.read_as_unbounded(env.in_len())?;
+                // Similar to EVM behavior, the `source` should be (limited to) the
+                // contract address. Otherwise contracts would be able to do arbitrary
+                // things on behalf of the caller via XVM.
+                let source = env.ext().address();
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The comment suggests that the `source` should be limited to the contract address for security reasons. However, the old code doesn't reflect this behavior. Please update the old code to use the contract address as the `source`.
```diff
-let _origin_address = env.ext().address().clone();
+let source = env.ext().address().clone();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->